File: IS (Java)/DES.java
import javax.crypto.*;
import javax.crypto.spec.SecretKeySpec;

public class DES {
    public static void main(String[] args) throws Exception {
        String msg = "Hello";
        byte[] keyBytes = "12345678".getBytes();
        SecretKey key = new SecretKeySpec(keyBytes, "DES");

        Cipher c = Cipher.getInstance("DES");
        c.init(Cipher.ENCRYPT_MODE, key);
        byte[] enc = c.doFinal(msg.getBytes());

        c.init(Cipher.DECRYPT_MODE, key);
        byte[] dec = c.doFinal(enc);

        for (byte b : enc) System.out.printf("%02x", b);
        System.out.println();
        System.out.println(new String(dec));
    }
}

File: IS (Python)/DES.py
from Crypto.Cipher import DES

msg = b"Hello".ljust(8, b" ")
key = b"12345678"

c = DES.new(key, DES.MODE_ECB)
enc = c.encrypt(msg)
dec = c.decrypt(enc)

print(enc.hex())
print(dec.decode().strip())
